polar/src/lexer.rs

```diff
@@ -340,19 +340,23 @@ impl<'input> Lexer<'input> {
 
         last = self.match_digits(last);
 
-        if let Some((i, '.')) = self.c {
-            self.push_char('.');
-            last = i;
-            parse_as_float = true;
-
-            last = self.match_digits(last);
+        if let Some((i, char)) = self.c {
```

Comment:
This isn't right anymore. Is this because you wanted to allow 1e4 without a `.`? Currently as this is it'd allow `1ee2` which isn't valid.

Reply:
added a fix